src/main/java/gov/usgs/owi/nldi/controllers/CharacteristicsController.java

```diff
@@ -75,4 +75,18 @@ public void getCharacteristicData(HttpServletRequest request, HttpServletRespons
 		}
 		logService.logRequestComplete(logId, response.getStatus());
 	}
+
+	@GetMapping(value="{featureSource}/{featureID}/basin")
+	public void getBasin(HttpServletRequest request, HttpServletResponse response,
+			@PathVariable(LookupDao.FEATURE_SOURCE) String featureSource,
+			@PathVariable(Parameters.FEATURE_ID) String featureID) throws IOException {
+		BigInteger logId = logService.logRequest(request);
+		try {
+			streamBasin(response, getComid(featureSource, featureID));
+		} catch (Throwable e) {
```

Comment:
Carry over from WQP - but I can't find the documentation on why it was changed there. This is (almost) the highest level of the code under our control. As such, it is the last place to prevent implementation details from escaping into the wild and to attempt an orderly cancel of the code. Definitely something we wouldn't want deeper in the logic.